packages/cactus-common/src/main/typescript/exception/exception-helper.ts

```diff
@@ -0,0 +1,327 @@
+import { RuntimeError } from "run-time-error";
+import stringify from "fast-safe-stringify";
+import { Logger } from "../logging/logger";
+
+export class ExceptionHelper {
+  /**
+   * Fetches the exception-content and submits it to the logger
+   *
+   * @param logger logger created thru the LoggerProvider.getOrCreate()
+   * @param exception as an arbitrary exception of whatever type and format it might be
+   * @param optionalAdditionalContext as optional on top info / context on the exception scenario
+   */
+  public static logException(
+    logger: Logger,
+    exception: unknown,
+    optionalAdditionalContext = "",
+  ): void {
+    const exceptionContent = this.getExceptionContent(
+      exception,
+      optionalAdditionalContext,
+    );
+
+    // log to destination
+    logger.error(...exceptionContent);
+  }
+
+  /**
+   * Creates a new RuntimeError based on the incoming exception and context to preserve the complete hierarchy of exception information
+   *
+   * @param exception as an arbitrary exception of whatever type and format it might be
+   * @param context providing context about the exception scenario
+   * @returns a new RuntimeError created from exception to preserve the complete hierarchy of exception information
+   */
+  public static getNewRuntimeErrorFromException(
```

Comment:
@m-courtin I recommend removing the `get` from the name because it instantiates/creates a new object instead of retrieving an existing one. Also recommend dropping `FromException` because we accept the `exception` input as anything (`unknown`)

Reply:
@petermetz: I'm fine with the renaming

Reply:
@m-courtin Thank you!